### DISCUSSION TITLE
how can i do equivalent of torch.cat in ggml tensor ?

### DISCUSSION THREAD

**Discussion by manmay-nakhashi** [ggml-org/ggml#421] [Q&A] [answered]
i want to cocatenate two tensor with dim,1024 how can i do that ?

**Comment by slaren** [comment]
I don't think there is an op to do this directly, but you could get the same effect by creating two views with different offsets and copying each part separately:
```C
// concat a,b into dst
dst = ggml_new_tensor_1d(ctx, a->type, a->ne[0] + b->ne[0]);
dst_lo = ggml_view_1d(ctx, dst, a->ne[0], 0);
dst_hi = ggml_view_1d(ctx, dst, b->ne[0], ggml_element_size(dst) * a->ne[0]);
ggml_cpy(ctx, a, dst_lo);
ggml_cpy(ctx, b, dst_hi);
```

**Reply by manmay-nakhashi** [reply]
and how can i do the same in 2d ?

**Reply by manmay-nakhashi** [reply, marked as answer]
```
        struct ggml_tensor * condL = ggml_new_tensor_2d(ctx0, cond->type, 1024, cond->ne[1] + textL->ne[1]);
        struct ggml_tensor * dst_lo = ggml_view_2d(ctx0, condL, 1024, cond->ne[1], cond->nb[1],0);
        struct ggml_tensor * dst_hi = ggml_view_2d(ctx0, condL, 1024, textL->ne[1], textL->nb[1], ggml_element_size(condL) * cond->ne[1]);
        ggml_cpy(ctx0, cond, dst_lo);
        ggml_cpy(ctx0, textL, dst_hi);
```

**Reply by manmay-nakhashi** [reply]
@slaren is this a right implementation ?

**Reply by slaren** [reply]
For `dst_hi` I think you want to set the offset to skip `cond->ne[1]` rows, but currently you are only skipping `cond->ne[1]` elements. This should work generally:
```C
struct ggml_tensor * dst_hi = ggml_view_2d(ctx0, condL, 1024, textL->ne[1], textL->nb[1], condL->nb[1] * cond->ne[1]);
```
